cwms-data-api/src/main/java/cwms/cda/api/BasinController.java

```diff
@@ -145,37 +179,168 @@ public void getOne(@NotNull Context ctx, @NotNull String basinId) {
                     Formats.NAMED_PGJSON;
             ContentType contentType = Formats.parseHeader(formatHeader, Basin.class);
             ctx.contentType(contentType.toString());
-            BasinDao basinDao = new BasinDao(dsl);
-            Basin basin = basinDao.getBasin(basinId, units, office);
+            String result;
+
             if (contentType.getType().equals(Formats.NAMED_PGJSON)) {
-                NamedPgJsonFormatter basinPgJsonFormatter = new NamedPgJsonFormatter();
-                String result = basinPgJsonFormatter.format(basin);
+                BasinDao basinDao = new BasinDao(dsl);
+                Basin basin = basinDao.getBasin(basinName, units, office);
+                result = Formats.format(contentType, Collections.singletonList(basin), Basin.class);
                 ctx.result(result);
+                ctx.status(HttpServletResponse.SC_OK);
             } else {
-                ctx.status(HttpServletResponse.SC_NOT_FOUND).json(new CdaError("Unsupported "
-                        + "format for basins"));
+                cwms.cda.data.dao.basin.BasinDao basinDao = new cwms.cda.data.dao.basin.BasinDao(dsl);
+                CwmsId basinId = new CwmsId.Builder()
+                        .withName(basinName)
+                        .withOfficeId(office)
+                        .build();
+                cwms.cda.data.dto.basin.Basin basin = basinDao.getBasin(basinId, units);
+                result = Formats.format(contentType, Collections.singletonList(basin),
+                        cwms.cda.data.dto.basin.Basin.class);
+                ctx.result(result);
+                ctx.status(HttpServletResponse.SC_OK);
             }
         } catch (SQLException ex) {
-            String errorMsg = "Error retrieving " + basinId;
+            CdaError error = new CdaError("Error retrieving " + basinName);
+            String errorMsg = "Error retrieving " + basinName;
             LOGGER.log(Level.SEVERE, errorMsg, ex);
+            ctx.status(HttpServletResponse.SC_INTERNAL_SERVER_ERROR).json(error);
         }
     }
 
-    @OpenApi(ignore = true)
+    @OpenApi(
+        responses = {
+            @OpenApiResponse(status = STATUS_404, description = "The provided combination of "
+                    + "parameters did not find a basin."),
+            @OpenApiResponse(status = STATUS_501, description = "Requested format is not "
+                    + "implemented")
+        },
+        description = "Renames CWMS Basin",
+        tags = {TAG}
+    )
     @Override
-    public void update(@NotNull Context ctx, @NotNull String s) {
-        ctx.status(HttpCode.NOT_IMPLEMENTED).json(CdaError.notImplemented());
+    public void update(@NotNull Context ctx, @NotNull String basinId) {
```

Comment:
Since this is a rename, we want to keep the new basin name as a query param. There's an existing NAME param in the Controllers class you can re-use. Do what you had before in this method, but just query that, instead of making a NEW_ID string.

Reply:
@adamkorynta - I think the reasoning behind using a query param for rename is so we don't need to build up an entire object just for a rename?

Reply:
Fixed in latest commit